### PR TITLE
Setup DesktopGL project

### PR DESCRIPTION
**Pull request by cra0zy**
Notes:
- Restored the csproj ignore to .gitignore, we should force csproj files until move is done
- Added DesktopGL project
  - This targets both `.netstandard2.0` and `.net451`, and this means that the nuget will be referencable by older projects as well
- Project file is a nuget package file at the same time :)
- I'm cheating because this isn't connected to the build bot, and is not breaking any stuff as our project name was never .DesktopGL.csproj

How to build nuget:
```sh
dotnet restore MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj
msbuild /t:Pack /p:Version=3.8.0.3 MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj
```

CC. @Jjagg

### CODE REVIEW

**Comment by mrhelmut** [comment]
```net4``` would be cool to have a compatibility toward WinXP, though it's not a big deal considering that Steam has stopped supporting it.
If it's holding anything back to move forward, let's just bid farewell to Net4.0 and WinXP (it has bad OpenGL drivers anyway).

**Comment by cra0zy** [comment]
Yea, I actually need .NET 4.5.1 actually for corert stuff.

**Comment by mrhelmut** [comment]
So goodbye WinXP, that's a problem solved in some way. =P
Can't wait for a smoother CoreRT support!

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    <Version>1.0.0.0</Version>
+    <Authors>cra0zy</Authors>
+    <PackageId>MonoGame.Framework.DesktopGL</PackageId>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <OutputPath>bin\DesktopGL\AnyCPU\Debug</OutputPath>
+    <IntermediateOutputPath>obj\Linux\AnyCPU\Debug</IntermediateOutputPath>
+    <DocumentationFile>bin\DesktopGL\AnyCPU\Debug\MonoGame.Framework.xml</DocumentationFile>
+    <DefineConstants>DEBUG;OPENGL;OPENAL;XNADESIGNPROVIDED;TRACE;LINUX;DESKTOPGL;SUPPORTS_EFX;NETSTANDARD</DefineConstants>
```
Comment (Jjagg) [inline MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: DEBUG is added automatically if the configuration is Debug.

We don't need two property groups here. Only difference is DocumentationFile and that can be set with `$(Configuration)` to differentiate between Debug and Release

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    <Compile Include="..\ThirdParty\NVorbis\NVorbis\Ogg\OggPacket.cs">
+      <Link>Platform\NVorbis\Ogg\OggPacket.cs</Link>
+    </Compile>
+    <Compile Include="..\ThirdParty\NVorbis\NVorbis\Ogg\OggPacketReader.cs">
+      <Link>Platform\NVorbis\Ogg\OggPacketReader.cs</Link>
+    </Compile>
+    <Compile Include="..\ThirdParty\NVorbis\NVorbis\Ogg\OggPageFlags.cs">
+      <Link>Platform\NVorbis\Ogg\OggPageFlags.cs</Link>
+    </Compile>
+    <Compile Include="Platform\GraphicsDeviceManager.SDL.cs" />
```
Comment (Jjagg) [inline MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: Can you put this with the other non-linked includes?

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 3]
```diff
@@ -0,0 +1,218 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Jjagg) [inline MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: Missing
```
    <EmbeddedResource Include="MonoGame.bmp">
      <LogicalName>MonoGame.bmp</LogicalName>
    </EmbeddedResource>
```
and
```
    <EmbeddedResource Include="..\ThirdParty\SDL_GameControllerDB\gamecontrollerdb.txt">
      <LogicalName>gamecontrollerdb.txt</LogicalName>
    </EmbeddedResource>
```

**Comment by Jjagg** [comment]
Building nugets can be 
```
dotnet restore MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj
dotnet pack -p:Version=3.8.0.3 MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj
```

File: MonoGame.Framework.DesktopGL.sln [hunk 4]
```diff
@@ -0,0 +1,34 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+VisualStudioVersion = 15.0.26124.0
+MinimumVisualStudioVersion = 15.0.26124.0
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MonoGame.Framework.DesktopGL", "MonoGame.Framework\MonoGame.Framework.DesktopGL.csproj", "{C01379FE-63B0-4DDC-AE9C-9C916CCE5F45}"
```
[the remark this answers is not in the document]
Comment (cra0zy) [reply MonoGame.Framework.DesktopGL.sln]: That uses .Linux.csproj project. We never had a .DesktopGL.csproj.

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <TargetFrameworks>netstandard2.0;net451</TargetFrameworks>
+    <RootNamespace>Microsoft.Xna.Framework</RootNamespace>
+    <AssemblyName>MonoGame.Framework</AssemblyName>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <PackageIconUrl>https://pbs.twimg.com/profile_images/487954549441691649/O3KsHAsb_400x400.png</PackageIconUrl>
+    <PackageProjectUrl>https://github.com/cra0zy/MonoGame/tree/core</PackageProjectUrl>
```
[the remark this answers is not in the document]
Comment (cra0zy) [reply MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: Ups. Copied over stuff from my .net core fork, missed that.

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+    <RepositoryUrl>https://github.com/cra0zy/MonoGame/tree/core</RepositoryUrl>
+    <Description>.NET Standard version of MonoGame DesktopGL platform.</Description>
+    <PackageTags>monogame;.net core;core;.net standard;standard;desktopgl</PackageTags>
+    <Version>1.0.0.0</Version>
```
[the remark this answers is not in the document]
Comment (cra0zy) [reply MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: When you pass it in the command line it get overwritten anyway.

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+    <Authors>cra0zy</Authors>
+    <PackageId>MonoGame.Framework.DesktopGL</PackageId>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <OutputPath>bin\DesktopGL\AnyCPU\Debug</OutputPath>
```
[the remark this answers is not in the document]
Comment (cra0zy) [reply MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: Only works if you specify it like that in the dir props file. There is a bug bout that.

File: MonoGame.Framework.DesktopGL.sln (hunk 4 above)
[the remark this answers is not in the document]
Comment (Jjagg) [reply MonoGame.Framework.DesktopGL.sln]: Okay, let's add them when we migrate those then.

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj (hunk 7 above)
[the remark this answers is not in the document]
Comment (Jjagg) [reply MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: You mean microsoft/msbuild#1603?

**Comment by cra0zy** [comment]
> dotnet pack

That only works on Windows, on Mac/Linux you have to use `msbuild` to pack multitarget framework nugets. TLDR. dotnet msbuild does not support standard .net framework compilation on non Windows platforms.

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 8]
```diff
@@ -0,0 +1,206 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <DefineConstants>OPENGL;OPENAL;XNADESIGNPROVIDED;TRACE;LINUX;DESKTOPGL;SUPPORTS_EFX;NETSTANDARD</DefineConstants>
```
Comment (Jjagg) [inline MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: `TRACE` can be omitted, it is added by default.

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+    <Content Include="..\ThirdParty\Dependencies\openal-soft\Windows\x64\soft_oal.dll" PackagePath="runtimes\win-x64\native" Visible="false" />
+    <Content Include="..\ThirdParty\Dependencies\SDL\Windows\x64\SDL2.dll" PackagePath="runtimes\win-x64\native" Visible="false" />
+    
+    <Content Include="..\ThirdParty\Dependencies\openal-soft\Linux\x64\libopenal.so.1" PackagePath="runtimes\linux-x64\native" Visible="false" />
+    <Content Include="..\ThirdParty\Dependencies\SDL\Linux\x64\libSDL2-2.0.so.0" PackagePath="runtimes\linux-x64\native" Visible="false" />
+    
+    <Content Include="..\ThirdParty\Dependencies\openal-soft\MacOS\Universal\libopenal.1.dylib" PackagePath="runtimes\osx\native" Visible="false" />
+    <Content Include="..\ThirdParty\Dependencies\SDL\MacOS\Universal\libSDL2-2.0.0.dylib" PackagePath="runtimes\osx\native" Visible="false" />
+
+    <Content Include="MonoGame.Framework.DesktopGL.targets" PackagePath="build" />
```
Comment (Jjagg) [inline MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: Why are these necessary? Is it to include only relevant files when building for a specific RID? And I suppose Visible=false because the ones below are shown already?

File: MonoGame.Framework/Directory.Build.props [hunk 10]
```diff
@@ -1,8 +1,15 @@
 <Project>
   <PropertyGroup>
-    <MonoGamePlatform>$(MSBuildProjectName.Substring(9))</MonoGamePlatform>
```
Comment (Jjagg) [inline MonoGame.Framework/Directory.Build.props]: Neat trick! Won't work for Net projects though. (but we should drop those anyway)

Comment (cra0zy) [reply MonoGame.Framework/Directory.Build.props]: You added that XD

File: MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj (hunk 9 above)
Comment (cra0zy) [reply MonoGame.Framework/MonoGame.Framework.DesktopGL.csproj]: Above ones get included in the nuget with very specific locations (based on RID) for .NET Core publish process.

The targets file is needed to copy them over when you use .NET Framework as target.

Bellow ones only get copied over when building project from source.

File: MonoGame.Framework/Directory.Build.props (hunk 10 above)
Comment (Jjagg) [reply MonoGame.Framework/Directory.Build.props]: 😅

**Comment by cra0zy** [comment]
@Jjagg should be good now, one thing we forgot is Framework.Net.Platform projects are still in the same folder, we should move it to a separate folder.

**Comment by Jjagg** [comment]
URLs in props file are still yours.

Yeah, we don't have to do Net stuff in this PR.

**Comment by cra0zy** [comment]
@Jjagg Yea, too many small mistakes to keep track of em. Hopefully no mistake now.

**Comment by Jjagg** [comment]
I tried building a game project created from the dotnet template with p2p reference to this csproj file and the MonoGamePlatform property didn't get set for my project (causing an error in MonoGame.Content.Builder).

**Comment by cra0zy** [comment]
> I tried building a game project created from the dotnet template with p2p reference to this csproj file and the MonoGamePlatform property didn't get set for my project (causing an error in MonoGame.Content.Builder).
> 
> EDIT: Maybe we should just add the property to the template directly? Anyway, that's not an issue with this PR.

Doesn't work like that, if you are not using nuget you have to manually set `MonoGamePlatform` property if you wanna use MGCB targets, thats normal.

**Comment by Jjagg** [comment]
Okay, this is good to go. Merging! :D
